### PR TITLE
Improve Slack Alerter by adding message blocks

### PR DESCRIPTION
**Pull request by soubenz**
## Describe changes
I added the support of blocks in the slack alerter component. Blocks allow further customization of the slack message, the current implementation will generate a slack message in the following format :

zenml_slack_bot

## Pre-requisites
Please ensure you have done the following:
- [X] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [X] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Comment by htahir1** [comment]
@soubenz Thanks for the contribution! Before we begin reviewing, could you make sure you resolve the conflicts? :-) Then we'll leave comments!

Again, super happy for your contribution!

**Comment by htahir1** [comment]
@soubenz Just a soft ping on this one :-)!

**Comment by soubenz** [comment]
> @soubenz Just a soft ping on this one :-)!

@htahir1  thanks! 
Conflicts are resolved now :)

**Comment by htahir1** [comment]
@soubenz Nice! Is it still a draft PR or would you mark it as ready for review now?

File: src/zenml/integrations/slack/alerters/slack_alerter.py [hunk 1]
```diff
@@ -131,6 +141,59 @@ def _get_disapprove_msg_options(
...
+        self, message: str, params: Optional[BaseAlerterStepParameters]
+    ) -> List[Dict]:  # type: ignore
+        """Define which messages will lead to disapproval during ask().
+
+        Args:
+            message: message
+            params: Optional parameters.
+
+        Returns:
+            Set of messages that lead to disapproval in alerter.ask().
```
Comment (strickvl) [inline src/zenml/integrations/slack/alerters/slack_alerter.py:154]: As far as I can see, this block creation has nothing to do with disapproval during `ask()`. The docstring probably needs updating.

**Review by strickvl — CHANGES_REQUESTED** [review]
Some initial feedback following a quick review just now. Also noticed some linting errors showing up in the CI. Could you take a look at those?

Thank you for the contribution, though! This makes the alerter messages much easier on the eyes! 🎨

**Review by fa9r — CHANGES_REQUESTED** [review]
Awesome, love it!

The only change I'd request is to make the new formatting optional so other users can still configure their step to get the plain message version in case they prefer that.

Once that is possible I'd be happy to approve this and see it go live, super useful feature 👍

**Comment by fa9r** [comment]
Oh, also, if you like you could also add the same logic to the ask step. I think having the environment info is super useful as context for all messages actually.

File: src/zenml/integrations/slack/alerters/slack_alerter.py (hunk 1 above)
Comment (soubenz) [reply src/zenml/integrations/slack/alerters/slack_alerter.py]: done !

File: src/zenml/integrations/slack/alerters/slack_alerter.py [hunk 2]
```diff
@@ -44,6 +53,8 @@ class SlackAlerterParameters(BaseAlerterStepParameters):
 
     # Set of messages that lead to disapproval in alerter.ask()
     disapprove_msg_options: Optional[List[str]] = None
+    payload: Optional[SlackAlerterPayload]
```
Comment (fa9r) [inline src/zenml/integrations/slack/alerters/slack_alerter.py:56]: I guess this needs to have a default None to be optional

```suggestion
    payload: Optional[SlackAlerterPayload] = None
```

**Review by fa9r — APPROVED** [review]
Awesome, that was quick. Looks good to me, thanks again for contributing 🙏 🚀

**Comment by strickvl** [comment]
Thank you for the updates and pushed changes. I'll just let the CI tests run through but otherwise it looks good to go from my side. 🙏🏻🙏🏻

**Pull request merged**